lib/Adapter/WorseReflection/Refactor/WorseExtractMethod.php

```diff
@@ -263,10 +264,10 @@ private function addParametersAndGetArgs(array $freeVariables, MethodBuilder $me
             }
 
             $parameterBuilder = $methodBuilder->parameter($freeVariable->name());
-            $variableType = $freeVariable->symbolContext()->type();
-
+            $variableType = $freeVariable->symbolContext()->types()->best();
             if ($variableType->isDefined()) {
-                $parameterBuilder->type($variableType->short());
+                $prefix = $variableType->isNullable() ? '?' : '';
```

Comment:
this should rather be done here I think: https://github.com/phpactor/code-builder/blob/master/templates/MethodHeader.php.twig#L14